### PR TITLE
Allowed building a HB with a non-0 starting epoch

### LINKED ISSUE #275: Allow starting a new network in an epoch other than 0.
E.g. when using Honey Badger for a blockchain, and the network is shut down and restarted, we'd probably want to start with a later epoch, to keep the epoch number in sync with the block number.

### PR DESCRIPTION
**Pull request by vkomenda**
This is a basic tool for starting `HoneyBadger` instances in non-trivial epochs as suggested by @afck in https://github.com/poanetwork/hbbft/issues/226#issuecomment-421761224.

Closes #275

### CODE REVIEW

**Comment by afck** [comment]
Sorry, I think that was a misunderstanding: We shouldn't ever need to start a Honey Badger instance in an epoch other than 0. Our _own_ epochs will never be skipped. What I meant by that comment was that the next Honey Badger instance should, at or immediately after its creation, handle any previously received `EpochStarted` messages concerning it (i.e. all that have the new `start_epoch` and therefore couldn't be handled previously) to update its knowledge of the _other_ nodes' current epochs.
(But maybe I'm just confused—please reopen if I'm wrong.)

**Comment by afck** [comment]
> We shouldn't ever need to start a Honey Badger instance in an epoch other than 0.

I strongly disagree with @afck here! :grin: 
See https://github.com/poanetwork/hbbft/issues/275.

File: src/honey_badger/builder.rs [hunk 1]
```diff
@@ -56,7 +66,7 @@ where
         };
         let step = if self.netinfo.is_validator() {
             // The first message in an epoch announces the epoch transition.
-            Target::All.message(Message::EpochStarted(0)).into()
+            Target::All.message(Message::EpochStarted(epoch)).into()
```
Comment (afck) [inline src/honey_badger/builder.rs]: I think we'd just need to remove this one line and rebase.
Then the DHB builder will need the same functionality.

**Comment by vkomenda** [comment]
Let me know if you want a `HoneyBadger` starting epoch to be configurable from the `DynamicHoneyBadger` builder. At the moment it isn't.

**Pull request merged**